From b13487a97b51a619b4e681a6091a1da9d6733058 Mon Sep 17 00:00:00 2001
From: Jason Liu <jason.hui.liu@nxp.com>
Date: Fri, 10 Jan 2020 11:38:05 +0800
Subject: [PATCH] LF-789-1 Revert "scripts: Makefile: Enable creation of
 _symbols_ DT node for overlays"

This reverts commit 432071b7a106060fa4e451e10249cef021af0b7c.

commit: 432071b7a106 scripts: Makefile: Enable creation of _symbols_ DT node for overlays
changes the common Makefile and force creation of __symbols__ node on all platforms.

This is not good and not acceptible under some cases due to this change will increase
the final DTB size a lot and bring big impact for others who does not need creation
of _symbols_ DT node for overlays.For example, on i.MX OP-TEE, the maxsize of DT is 1MB,
this patch will break some of the i.MX6/i.MX7 boards to boot with OP-TEE enabled.

BTW, community has the similar patch but rejected. The following post discuss about this:
https://lore.kernel.org/patchwork/patch/821645/

For specifc plaform/board which need the creation of _symbols_ DT node for overlays, user can
define DTC_FLAGS_target either trough dtc build command line(#1) or with board specific Makefile(#2)

For example:#1

make DTC_FLAGS_fsl-ls1028a-qds=-@ freescale/fsl-ls1028a-qds.dtb

For example:#2

In arch/arm64/boot/dts/freescale/Makefile,

 @@ -1,4 +1,7 @@
  # SPDX-License-Identifier: GPL-2.0
 +# required for overlay support
 +#DTC_FLAGS_fsl-ls1028a-qds := -@
 +

Signed-off-by: Jason Liu <jason.hui.liu@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Reviewed-by: Alex Marginean <alexandru.marginean@nxp.com>
Tested-by: Alex Marginean <alexandru.marginean@nxp.com>
[fix commit message]
Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
---
 scripts/Makefile.lib | 3 ---
 1 file changed, 3 deletions(-)

--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -255,9 +255,6 @@ endif
 
 DTC_FLAGS += $(DTC_FLAGS_$(basetarget))
 
-# Enable creation of _symbols_ node for DT overlays
-DTC_FLAGS += -@
-
 # Generate an assembly file to wrap the output of the device tree compiler
 quiet_cmd_dt_S_dtb= DTB     $@
 cmd_dt_S_dtb=						\
